Projects/Feature/FilterFeature/Sources/Scene/FilterView.swift

```diff
@@ -0,0 +1,456 @@
+    var intent: any FilterIntentProtocol { container.intent }
+    var state: any FilterStateProtocol { container.model }
+    private let techStackAppendBuildable: any TechStackAppendBuildable
+    let columns = [GridItem(.flexible()), GridItem(.flexible())]
+
+    init(
+        container: MVIContainer<FilterIntentProtocol, FilterStateProtocol>,
+        techStackAppendBuildable: any TechStackAppendBuildable
+    ) {
+        self._container = StateObject(wrappedValue: container)
+        self.techStackAppendBuildable = techStackAppendBuildable
+    }
+
+    var body: some View {
+        GeometryReader { geometry in
+            ZStack(alignment: .bottom) {
+                ScrollView(showsIndicators: false) {
+                    SMSSeparator()
+                        .padding(.bottom, 20)
+
+                    VStack(alignment: .leading, spacing: 40) {
+                        ConditionView(state.userRole == .guest) {
+                            checkBoxSection(
+                                text: "학년",
+                                isSelected: {
+                                    state.gradeSet.contains($0)
+                                },
+                                selectAction: {
+                                    if state.gradeSet.contains($0) == false {
+                                        intent.gradeSelect(grade: $0)
+                                    } else {
+                                        intent.gardeDeSelect(grade: $0)
+                                    }
+                                },
+                                data: GradeType.allCases,
+                                id: \.self
+                            ) { text in
+                                SMSText("\(text.rawValue)학년", font: .body2)
+                            }
+
+                            checkBoxSection(
+                                text: "반",
+                                isSelected: {
+                                    state.classSet.contains($0)
+                                },
+                                selectAction: {
+                                    if state.classSet.contains($0) == false {
+                                        intent.classSelect(class: $0)
+                                    } else {
+                                        intent.classDeSelect(class: $0)
+                                    }
+                                },
+                                data: ClassType.allCases,
+                                id: \.self
+                            ) { text in
+                                SMSText("\(text.rawValue)반", font: .body2)
+                            }
+
+                            checkBoxSection(
+                                text: "학과",
+                                isSelected: {
+                                    state.departmentSet.contains($0)
+                                },
```

Comment:
```suggestion
                                isSelected: state.departmentSet.contains,
```
로 되긴한데 혹시 이건 어떻게 생각하시나요? 호불호 조금 갈릴거같아서요,